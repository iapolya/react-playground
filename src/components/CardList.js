import React from "react";
import Card from "./Card";

const CardList = ({cats}) => {
  return (
    <div>
      {cats.map((cat, i) => {
        return (
          <Card key={i} id={cat.id} name={cat.name} email={cat.email}/>
        )
      })}
    </div>
  )
}

export default CardList;