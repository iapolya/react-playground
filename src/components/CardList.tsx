import React from 'react';
import Card from './Card';
import { ICat } from '../types';

const CardList = ({ cats }: { cats: Array<ICat> }) => {
  return (
    <section>
      {cats.map((cat, i) => {
        return <Card key={i} id={cat.id} name={cat.name} email={cat.email} />;
      })}
    </section>
  );
};

export default CardList;
