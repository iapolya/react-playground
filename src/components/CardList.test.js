import React from 'react';
import { shallow } from 'enzyme';
import CardList from './CardList';

it('expect to render card list component', () => {
  const mockCats = [
    {
      id: 1,
      name: 'name1',
      email: 'email1'
    },
    {
      id: 2,
      name: 'name2',
      email: 'email2'
    }
  ];
  expect(shallow(<CardList cats={mockCats} />)).toMatchSnapshot();
});
