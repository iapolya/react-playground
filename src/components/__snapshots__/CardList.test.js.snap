// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render card list component 1`] = `
<section>
  <Card
    email="email1"
    id={1}
    key="0"
    name="name1"
  />
  <Card
    email="email2"
    id={2}
    key="1"
    name="name2"
  />
</section>
`;
